scripts/extract_hh.py

```diff
@@ -1,69 +1,47 @@
-#! /usr/bin/env python
+#!/usr/bin/env python
 # -*- coding: utf-8 -*-
-# Copyright 2018 the HERA Project
+# Copyright 2019 The HERA Collaboration
 # Licensed under the MIT License
 
+from __future__ import print_function, division, absolute_import
 import os
-import argparse
-import pyuvdata
-from hera_cal.version import version_info
-import pyuvdata.utils as uvutils
+import sys
 import numpy as np
+import argparse
+from pyuvdata import UVData
+from pyuvdata.uvh5 import _hera_corr_dtype
+
+# create an argument parser and parse args
+ap = argparse.ArgumentParser(description="Extract only connected antennas from full correlator data")
+ap.add_argument("input_file", type=str, help="Full path to target file")
+ap.add_argument("output_file", type=str, help="Full path to output file")
+ap.add_argument("--overwrite", action="store_true", default=False,
+                help="Optional flag to overwrite output file if it exists")
+args = ap.parse_args()
+
+# read in file metadata
+uvd = UVData()
+fn_in = args.input_file
+fn_out = args.output_file
+if os.path.exists(fn_out) and not args.overwrite:
+    print("skipping {}...".format(fn_in))
+    sys.exit(0)
+print("scanning {}...".format(fn_in))
+uvd.read_uvh5(fn_in, read_data=False, run_check=False)
 
-parser = argparse.ArgumentParser(description='Extract HERA hex antennas from data '
-                                 'file, and save with new extension.')
-parser.add_argument('--extension', type=str, help='Extension to be appended to '
-                    'filename for output. Default="HH".', default='HH')
-parser.add_argument('--filetype', type=str, help='Input and output file type. '
-                    'Allowed values are "miriad" (default), and "uvfits".',
-                    default='miriad')
-parser.add_argument('--fixuvws', action='store_true', help='Optional flag to '
-                    'use antenna positions to replace uvws.')
-parser.add_argument('--overwrite', action='store_true', default=False,
-                    help='Optional flag to overwrite output file if it exists.')
-parser.add_argument('files', metavar='files', type=str, nargs='+',
-                    help='Files to be processed.')
-parser.add_argument('--ex_ants_file', type=str, help='Text file with list of antennas'
-                    ' which are excluded downstream in RTP. Generally, these are '
-                    'antennas which are being actively commissioned, or known as bad.'
-                    ' Note these values are only stored in the history, not actually '
-                    'flagged at this step.',
-                    default=None)
-args = parser.parse_args()
+# figure out which antennas have valid data and perform select-on-read
+ant_nums = np.unique(uvd.ant_1_array)
+inds = np.where(ant_nums < 350)
```

Comment:
The number 350 seems like it should be an optional command line argument.

Reply:
I suppose I could add it, but the way that Jack is writing correlator files right now, all antennas with actual data in them have numbers less than 350.